### PR TITLE
Feature/CO-1727 list documents

### PR DESCRIPTION
**Pull request by fernamig**

### CODE REVIEW

File: OnBaseDocsApi/Models/ParamCollection.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            if (value != null)
+                return long.Parse(value);
+            else
+                return defValue;
+        }
+
+        public string Get(string name, string defValue)
+        {
+            var value = Params[name];
+            if (value != null)
```
Comment (Ruefa) [inline OnBaseDocsApi/Models/ParamCollection.cs:36]: should we use IsNullOrEmpty(value) here?

File: OnBaseDocsApi/Controllers/OnBaseDocsController.cs [hunk 2]
```diff
@@ -189,6 +194,77 @@ public async Task<IHttpActionResult> Post()
...
         }
 
+        // We can't use [FromUri] to auto bind the query parameters because
+        // the binding does not work with '[' and ']'. So we must manually
+        // bind them.
+        [HttpGet]
+        [ActionName("")]
+        public IHttpActionResult ListDocs()
+        {
+            var parms = new ParamCollection(Request.RequestUri.ParseQueryString());
```
[the remark this answers is not in the document]
Comment (fernamig) [reply OnBaseDocsApi/Controllers/OnBaseDocsController.cs:204]: params is a reserved word, so I can't use that as a variable.

File: OnBaseDocsApi/Models/ParamCollection.cs (hunk 1 above)
Comment (fernamig) [reply OnBaseDocsApi/Models/ParamCollection.cs:36]: Good point.

File: OnBaseDocsApi/Controllers/OnBaseDocsController.cs [hunk 3]
```diff
@@ -189,6 +194,77 @@ public async Task<IHttpActionResult> Post()
...
+                if (queryResults == null)
+                    return InternalErrorResult("Document query returned null.");
+
+                var docs = new List<DataResource<DocumentAttributes>>();
+                foreach (var doc in queryResults)
+                {
+                    docs.Add(DocumentResource(doc));
+                }
+
+                // Generate the query string for this request.
```
[the remark this answers is not in the document]
Comment (fernamig) [reply OnBaseDocsApi/Controllers/OnBaseDocsController.cs:245]: Yeah, another good point. I moved the code to it's a builder class.

**Pull request merged**